### PR TITLE
Command to reload Nuvotifier configurations in Sponge

### PR DESCRIPTION
**Pull request by curscascis**
Hello. I hope this pull request is not much of a burden. I did not see any PR policies in the README. I would have built and published this myself under the same name - (except modified)  but it seems the licensing prohibits me from doing that, to my understanding. and so I'd like to contribute this bit of code which adds a command to NuVotifier (sponge Version) to destroy the previous channels and listening server and to create a new one with the newly loaded configuration.

This is an important change because often times when server owners do start up their voting processes they generally have players online. Having to restart servers over and over again as they troubleshoot here and there, working with server hosts and friends trying to figure out the right setup. All this rebooting is pretty annoying for players. Some servers end up changing dns records, or have ip  or port changes. At the end of the day it's intent is to improve the experience of the players. It's a very useful and needed tool and I am sure it will be used frequently among server owners.

### CODE REVIEW

**Comment by curscascis** [comment]
I have tested the functionality, and it works as intended.

File: sponge/src/main/java/com/vexsoftware/votifier/sponge/VotifierPlugin.java [hunk 1]
```diff
@@ -69,7 +77,41 @@ public VotifierPlugin(Logger logger) {
     public void onServerStart(GameStartedServerEvent event) {
         // Set the plugin version.
         version = this.getClass().getAnnotation(Plugin.class).version();
+        reloadConfig();
```
Comment (Ichbinjoe) [inline sponge/src/main/java/com/vexsoftware/votifier/sponge/VotifierPlugin.java]: You make this boolean, yet never bother to look to see what it returns?

File: sponge/src/main/java/com/vexsoftware/votifier/sponge/VotifierPlugin.java [hunk 2]
```diff
@@ -208,8 +250,7 @@ public void onServerStart(GameStartedServerEvent event) {
 
         serverGroup = new NioEventLoopGroup(1);
 
-        new ServerBootstrap()
-                .channel(NioServerSocketChannel.class)
+        new ServerBootstrap().channel(NioServerSocketChannel.class)
```
Comment (Ichbinjoe) [inline sponge/src/main/java/com/vexsoftware/votifier/sponge/VotifierPlugin.java]: Why did you make this one line? Unnecessary change.

**Comment by Ichbinjoe** [comment]
Are you able to write a bukkit api/bungee equivalent? I would want a feature like this to be on all versions.

If not that is fine - I may get to it in my spare time... (what even is spare time...)

File: sponge/src/main/java/com/vexsoftware/votifier/sponge/VotifierPlugin.java (hunk 2 above)
Comment (curscascis) [inline sponge/src/main/java/com/vexsoftware/votifier/sponge/VotifierPlugin.java]: I believe that was an accidental change, I was first trying to go about it a different way.  I haven't worked with bukkit/spigot API in a while and I have not worked with bungee plugins at all.  Since everything is already setup on my end I wouldn't mind taking a look and adding it, I don't have servers for those right now to test my changes though.

**Comment by curscascis** [comment]
I've made the changes as requested, and added the changes to Bungee and Bukkit. I have not been able to test those two, but I followed their formatting for adding commands. Since it follows the same format no issues should arise, but a testing is still warranted.

File: universal/pom.xml [hunk 3]
```diff
@@ -9,7 +9,7 @@
     </parent>
     <modelVersion>4.0.0</modelVersion>
 
-    <artifactId>nuvotifier-universal</artifactId>
+    <artifactId>serious-nuvotifier-universal</artifactId>
```
Comment (Ichbinjoe) [inline universal/pom.xml]: revert

File: sponge/pom.xml [hunk 4]
```diff
@@ -42,20 +42,30 @@
...
         </plugins>
     </build>
 
+
+
     <dependencies>
         <dependency>
             <groupId>com.vexsoftware</groupId>
             <artifactId>nuvotifier-common</artifactId>
-            <version>${project.parent.version}</version>
```
Comment (Ichbinjoe) [inline sponge/pom.xml]: why was this changed?

File: bungeecord/src/main/java/com/vexsoftware/votifier/bungee/commands/ReloadCommand.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        super("nvreload");
+        this.plugin = plugin;
+    }
+
+
+    @Override
+    public void execute(CommandSender commandSender, String[] strings) {
+        if(commandSender.hasPermission("nuvotifier.reload")){
+            if ((plugin.reloadConfigs())) {
+                commandSender.sendMessage(new ComponentBuilder("nuvotifier has been reloaded").color(ChatColor.GREEN).create());
```
Comment (Ichbinjoe) [inline bungeecord/src/main/java/com/vexsoftware/votifier/bungee/commands/ReloadCommand.java]: Odd capitalization?

File: bukkit/src/main/java/com/vexsoftware/votifier/NuVotifierBukkit.java [hunk 6]
```diff
@@ -283,20 +315,21 @@ public void operationComplete(ChannelFuture future) throws Exception {
...
-            if (serverChannel != null)
-                serverChannel.close();
-            serverGroup.shutdownGracefully();
-        }
-        if (forwardingMethod != null) {
-            forwardingMethod.halt();
+    public boolean onCommand(CommandSender sender, Command cmd, String commandLabel, String[] args){
+        if(cmd.getName().equalsIgnoreCase("nvreload")){
+            if(reloadConfigs()){
+                sender.sendMessage("NuVotifier Has reloaded Successfully");
```
Comment (Ichbinjoe) [inline bukkit/src/main/java/com/vexsoftware/votifier/NuVotifierBukkit.java]: Green color for consistency with sponge

**Comment by Ichbinjoe** [comment]
Sorry this took so long. I missed the ping for this :(
